pkg/kudoctl/packages/verifier/template/verify_render.go

```diff
@@ -18,9 +20,31 @@ func (RenderVerifier) Verify(pf *packages.Files) verifier.Result {
 }
 
 func templateCompilable(pf *packages.Files) verifier.Result {
-	params := make(map[string]string)
+	res := verifier.NewResult()
+
+	params := make(map[string]interface{}, len(pf.Params.Parameters))
+
 	for _, p := range pf.Params.Parameters {
-		params[p.Name] = "default"
+		switch p.Type {
+		case kudov1beta1.ObjectValueType:
+			value, err := convert.YAMLObject(convert.StringValue(p.Default))
+			if err != nil {
+				res.AddErrors(err.Error())
+			}
+
+			params[p.Name] = value
+		case kudov1beta1.ArrayValueType:
+			value, err := convert.YAMLObject(convert.StringValue(p.Default))
+			if err != nil {
+				res.AddErrors(err.Error())
+			}
+
+			params[p.Name] = value
+		case kudov1beta1.StringValueType:
+			fallthrough
+		default:
+			params[p.Name] = convert.StringValue(p.Default)
+		}
```

Comment:
I think this could be joined with `paramValue` from above, it's basically:

```
_, err := paramValue( &p.Default, p.Type )
if err != nil {
  res.AddErrors(err.Error())
}
```

Or am I missing something?

Reply:
Good point, let me add a `paramValue` function in the `convert` package.